greedy_algorithms/best_time_to_buy_and_sell_stocks_1.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+Best Time to Buy and Sell Stock - Greedy
+
+Problem Statement: 
+Given an array for which the ith element is the price of a given stock on day i.
+If you were only permitted to complete at most one transaction 
+(i.e., buy one and sell one share of the stock), design an algorithm to find the maximum profit.
+Note that you cannot sell a stock before you buy one.
+
+Solution:
+
+we find the maximum price for which the stock can be sell out at ith day in vector b, ith element of b
+is the maximum stock price from (n-1)th day to ith day. Then for each day find the profit by subtracing 
+the cost of buying at ith day, final answer will be the maximum of these profits.
+*/
+
+
+#include <iostream>
+#include <vector> 
+#include <cassert>
+
+int maxProfit(std::vector<int> prices, int n) 
+{
+    int i = 0, ans = 0;
+
+	// no profit if the number of days is zero
+    if(n == 0)
+	  {
+       return 0;
```

Comment:
the formatting seems a bit off, please use the recommended [clang-formatter](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#code-formatter) to follow the unified coding style. I also recommend setting up GitHub actions which automatically fixes the formatting issues for you.